src/BcPrebidVast.js

```diff
@@ -636,11 +641,76 @@ function loadMolPlugin(callback) {
     }
 }
 
+function getAdRendererFromAdOptions(adOptions) {
+	// if adRenderer option is present use it for all ads
+	if (adOptions.hasOwnProperty('adRenderer') && adOptions.adRenderer &&
+		(adOptions.adRenderer === _rendererNames.MOL || adOptions.adRenderer === _rendererNames.IMA)) {
+		return {adRenderer: adOptions.adRenderer, needBreak: true};
+	}
+	// for DFP use IMA plugin as renderer
+	if (adOptions.hasOwnProperty('dfpParameters')) {
+		return {adRenderer: _rendererNames.IMA, needBreak: false};
+	}
+	return null;
+}
+
+function setAdRenderer(options) {
+	if (options) {
+		var adRenderer = _rendererNames.MOL;
+		var rendObj;
+		var i;
+		if (Array.isArray(options)) {
+			for (i = 0; i < options.length; i++) {
+				rendObj = getAdRendererFromAdOptions(options[i]);
+				if (rendObj) {
+					adRenderer = rendObj.adRenderer;
+					if (rendObj.needBreak) {
+						break;
+					}
+				}
+			}
+			if (adRenderer) {
```

Comment:
Can you add a comment here to explain that if we did find a renderer this loop is there to set the same renderer across all configurations in the array?

Reply:
Made code optimization.